src/phoenix/server/app.py

```diff
@@ -657,11 +702,13 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
     )
     app.state.read_only = read_only
     app.state.export_path = export_path
+    app.state.validate_api_key = api_key_validator
     app.include_router(v1_router)
     app.include_router(router)
     app.include_router(graphql_router)
     app.add_middleware(GZipMiddleware)
-    if serve_ui:
+    web_manifest_path = SERVER_DIR / "static" / ".vite" / "manifest.json"
+    if serve_ui and web_manifest_path.is_file():
```

Comment:
This is just to make our lives a little easier: we don't need to toggle ui on and off when running integrations tests on github or locally: when running locally, we can pause the test via a break point and go checkout the app on the browser.